doc: add path aliases typescript doc (#55766)
nodejs/node: doc/api/typescript.md

Node.js will by default refuse to handle TypeScript files inside folders under
a `node_modules` path.

### Paths aliases

> [Path aliases][] won't be transformed and therefore produce an error. The closest

Reviewer: ```suggestion
[`tsconfig` "paths"][] won't be transformed and therefore produce an error. The closest
```

Reviewer: btw, @AugustinMauroy and I will be soon publish a loader that will handle this: https://github.com/JakobJingleheimer/nodejs-loaders

(it's currently available on npm, but currently you have to install all the loaders, so it'll be move into a namespace and installable individually, so `npm i @nodejs-loaders/aliases`). I'll update this doc subsequently to mention something about it.

Reviewer 2: why not in amaro?

Reviewer: I suppose it could. `@nodejs-loaders` contains more than just the aliases loader (it also has a css-modules loader, a tsx loader, etc).

Reviewer 3: > why not in amaro?

in amaro but not in node.js core ? I didn't get your idea

Reviewer 4: FWIW handling this may not be a good idea in general when tsc will not / cannot reason about them for its own emit in the rewriting; paths aliases are really not supposed to be used like this. package.json import aliases really seem like the right thing to suggest here.

Author: I think I'm missing something. Isn't that what we're suggesting? Typescript wouldn't rewrite the subpath imports and that would still work.

Anyway I might be missing something but using Node's typescript conversion and tsc at the same time doesn't seem possible, as for the first you need the `.ts` extension and the other won't deal with it.

Reviewer: > I think I'm missing something. Isn't that what we're suggesting? Typescript wouldn't rewrite the subpath imports and that would still work.

That is what I understood

Reviewer 4: Sure, it's just that it's code that will never run unless a loader fixes it, whereas the existing "run TS" support is more or less "not a loader" as it's just running the TS as-is and the same code can be pumped through tsc and still execute as JS. The same can't be said for tsconfig paths, which will crash without an extra loader that users of packages won't have installed.

> feature available are [subpath imports][] with the limitation that they need to start

Reviewer 4: ```suggestion
feature available is [subpath imports][] with the limitation that they need to start
```

(I think this is grammatically correct? otherwise `are` mismatches `closest feature` being singular...)

> with `#`

Reviewer 4: ```suggestion
with `#`.
```
